File: examples/android/app/src/test/java/org/aetherfeed/app/news/NewsFormatTest.kt
package org.aetherfeed.app.news

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class NewsFormatTest {
    @Test
    fun stripsHtmlAndCapsSnippet() {
        val long = "<p>" + "word ".repeat(80) + "</p>"
        val snippet = plainSnippet(long, 40)
        assertTrue(snippet.startsWith("word"))
        assertTrue(snippet.endsWith("…"))
        assertTrue("<" !in snippet)
    }

    @Test
    fun listSnippetDropsLinkStubs() {
        assertEquals("", listSnippet("<p>Article URL: https://ex.example/a</p>"))
        assertEquals("Hello world", listSnippet("<p>Hello world</p>"))
    }

    @Test
    fun agesMatchReadYouStyleBuckets() {
        val now = 100_000_000L
        assertEquals(AgeUnit.Empty, ageLabel(null, now).unit)
        assertEquals(AgeUnit.Now, ageLabel(now - 10 * 60_000L, now).unit)
        assertEquals(AgeLabel(AgeUnit.Hours, 5), ageLabel(now - 5 * 3_600_000L, now))
        assertEquals(AgeLabel(AgeUnit.Days, 3), ageLabel(now - 3 * 86_400_000L, now))
    }
}

File: examples/web/src/news/newsFormat.test.ts
import { describe, expect, it } from "vitest";
import { ageLabel, listSnippet, plainSnippet } from "./newsFormat";

describe("newsFormat", () => {
  it("strips html and caps the snippet", () => {
    const snippet = plainSnippet(`<p>${"word ".repeat(80)}</p>`, 40);
    expect(snippet.startsWith("word")).toBe(true);
    expect(snippet.endsWith("…")).toBe(true);
    expect(snippet.includes("<")).toBe(false);
  });

  it("drops link stubs from list snippets", () => {
    expect(listSnippet("<p>Article URL: https://ex.example/a</p>")).toBe("");
    expect(listSnippet("<p>Hello world</p>")).toBe("Hello world");
  });

  it("buckets age like ReadYou/Feeder metadata", () => {
    const now = 100_000_000;
    expect(ageLabel(undefined, now)).toBe("");
    expect(ageLabel(now - 10 * 60_000, now)).toBe("now");
    expect(ageLabel(now - 5 * 3_600_000, now)).toBe("5h");
    expect(ageLabel(now - 3 * 86_400_000, now)).toBe("3d");
  });
});
